### PR TITLE
[REF] report: pdf generation refactoring

### PR DESCRIPTION
**Pull request by qdp-odoo**
on behalf of @oco-odoo

### CODE REVIEW

File: addons/report/models/report.py [hunk 1]
```diff
@@ -142,11 +142,34 @@ def get_html(self, docids, report_name, data=None):
...
-        """This method generates and returns pdf version of a report.
         """
-
+        This method generates and returns pdf version of a report. It also generate 
+        the corresponding attachment.
+        """
+        attachment_callback_data = {}
+        pdf_report = self._generate_pdf_report(docids,report_name, html=html, data=data, report_attachment_callback_data=attachment_callback_data)
+        
+        if attachment_callback_data: #If values were inserted into the dictionary
```
Comment (qdp-odoo) [inline addons/report/models/report.py]: on line 152 you assign an empty dict on this variable and it is never set to something else, so the whole if statement will always be ignored

Comment (qdp-odoo) [inline addons/report/models/report.py]: wait a minute... are you using mutable dict?

**Comment by qdp-odoo** [comment]
anyway, let's forget about it

**Comment by Yenthe666** [comment]
@qdp-odoo why forget about it? It would be amazing to have extra flexibility to combine PDF's etc.

**Comment by qdp-odoo** [comment]
tldr: yeah, a refactoring of report creation will probably be done but in a much deeper way. 

This PR was proposed after an aborted attempt of an unrelated task that will finally be implemented in another way... I mean: the aim of the task was not to refactor the reports, and we just wanted to see if you could save something from this but it needs a more in-depth analysis/refactoring. For example to avoid the use of mutable dict that is not readable, and makes a normal behavior looks like a side-effect.... :-)

**Comment by lmignon** [comment]
@qdp-odoo For the record: For one of our Enterprise client we also need a way to post process the generated report. Here it's  my proposal https://github.com/odoo/odoo/pull/15091

**Comment by qdp-odoo** [comment]
@sle-odoo FYI ;-)
